assayist/processor/loose_artifact_analyzer.py

```diff
@@ -0,0 +1,125 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+import glob
+import os
+
+from itertools import zip_longest
+from hashlib import md5
+
+from assayist.common.models import content
+from assayist.processor.base import Analyzer
+from assayist.processor.logging import log
+
+
+class LooseArtifactAnalyzer(Analyzer):
+    """
+    Analyze RPMs/jars/similar that are embedded in the build artifacts.
+
+    RPMs we have to treat differently from other artifact types because Koji treats them
+    differently and there is no way to look up an rpm by checksum in the Koji api. So,
+    for RPMs we will look them up by filename, which should be uniquely identifying across
+    a single Koji instance (assuming someone hasn't renamed the file).
+
+    The intention for other artifact types (jar, pom, etc) is that we will checksum them
+    and then look them up in koji by checksum. You can accomplish the same thing with
+    koji-build-finder, but for o ur use-case doing it outselves was just as easy and made
+    dependency management and debugging a lot simpler.
+    """
+
+    FILE_EXTENSIONS = ['rpm', 'zip', 'tar', 'jar', 'pom.xml', 'pom', 'war']
```

Comment:
Let's add the rest of the tar and jar-related extensions as well:
* `tar tar.gz tar.bz2 tar.xz`
* `jar war rar ear sar kar jdocbook jdocbook-style plugin`

Reply:
Will you recurse into e.g. war files to look for jars ?

Reply:
@rnc No, we will create up a stub entry in the DB for an archive that exists in Koji and a successive analysis of that entry will look into the contents of that archive.

Reply:
@mprpic added the extra extensions in the latest version.